wp/lib/bap_wp/src/compare.mli

```diff
@@ -30,74 +34,93 @@ module Env = Environment
   -> output:Bap.Std.Var.Set.t
   -> original:(Bap.Std.Blk.t * Env.t)
   -> modified:(Bap.Std.Blk.t * Env.t)
   -> smtlib_post:string
   -> smtlib_hyp:string
   -> Constr.t * Env.t * Env.t
 
+(** Compare two subroutines by composition: given the lists of postconditions
+    and hypotheses generated by the API below. *)
+val compare_subs
+  :  postconds:(comparator list)
+  -> hyps:(comparator list)
+  -> original:(Bap.Std.Sub.t * Env.t)
+  -> modified:(Bap.Std.Sub.t * Env.t)
+  -> Constr.t * Env.t * Env.t
+
 (** Compare two subroutines by composition for equality of return
     values:
 
-    Given a set of common input and output variables, return a
+    Given a set of common input and output variables, return a postcondition
+    and hypothesis that, when passed to [compare_subs], will generate a
     precondition which is provable only if (modulo soundness bugs) the
     subroutines have equal output variables given equal input
     variables. *)
 val compare_subs_eq
   :  input:Bap.Std.Var.Set.t
   -> output:Bap.Std.Var.Set.t
-  -> original:(Bap.Std.Sub.t * Env.t)
-  -> modified:(Bap.Std.Sub.t * Env.t)
-  -> smtlib_post:string
-  -> smtlib_hyp:string
-  -> Constr.t * Env.t * Env.t
+  -> comparator * comparator
 
 (** Compare two subroutines by composition for an empty postcondition:
 
-    Given two subroutines and environments, return a
+    Given two subroutines and environments, return a postcondition
+    and hypothesis that, when passed to [compare_subs], will generate a
     precondition which is provable only if (modulo soundness bugs) the VCs generated
     by the hooks provided by the environment are satisfied. *)
-val compare_subs_empty
-  :  original:(Bap.Std.Sub.t * Env.t)
-  -> modified:(Bap.Std.Sub.t * Env.t)
-  -> smtlib_post:string
-  -> smtlib_hyp:string
-  -> Constr.t * Env.t * Env.t
+val compare_subs_empty : unit -> comparator * comparator
 
 (** Compare two subroutines by composition for an empty
     postcondition:
 
-    Given two subroutines and environments, return a precondition
-    which is provable only if (modulo soundness bugs), for equal
+    Given two subroutines and environments, return a postcondition
+    and hypothesis that, when passed to [compare_subs], will generate a
+    precondition which is provable only if (modulo soundness bugs), for equal
     inputs, the VCs generated by the hooks provided by the environment
     are satisfied. *)
-val compare_subs_empty_post
-  :  input:Bap.Std.Var.Set.t
-  -> original:(Bap.Std.Sub.t * Env.t)
-  -> modified:(Bap.Std.Sub.t * Env.t)
-  -> smtlib_post:string
-  -> smtlib_hyp:string
-  -> Constr.t * Env.t * Env.t
+val compare_subs_empty_post : unit -> comparator * comparator
+
+(** Compare two subs by composition for an empty postcondition:
+
+    Give two subroutines and environments, return a postcondition
+    and hypothesis that, when passed to [compare_subs], will generate a
+    precondition which is provable only if (module soundness bugs) the VCs generated
```

Comment:
module -> modulo

Reply:
Really everywhere. That's probably my typo, sorry.

Reply:
It might be worth elaborating on this comment, I'm honestly not sure what this comparator does, myself.